tdrs-backend/tdpservice/parsers/parse.py

```diff
@@ -276,7 +279,8 @@ def parse_datafile_lines(datafile, program_type, section, is_encrypted):
 
     # Only checking "all_created" here because records remained cached if bulk create fails. This is the last chance to
     # successfully create the records.
-    all_created, unsaved_records = bulk_create_records(unsaved_records, line_number, header_count, datafile, flush=True)
+    all_created, unsaved_records = bulk_create_records(unsaved_records, line_number, header_count, datafile, dfs,
+                                                       flush=True)
     if not all_created:
         logger.error(f"Not all parsed records created for file: {datafile.id}!")
         rollback_records(unsaved_records, datafile)
```

Comment:
if there's an error, after rollback, can we set `dfs.total_number_of_records_in_file` (and/or `dfs.total_number_of_records_created`) to `0`?